### ISSUE TITLE
Can't replace document's property with null

### ISSUE THREAD

**Issue by Try4W** [dizitart/nitrite-database#56]
When I trying to upsert object to DB:
```
caObjectsDao.put(caObject.copy(
    output2Status = CaoStatus(false, currentDateTime),
    output2TriggerEndDateTime = null
))
```

`output2Status` changed, but `output2TriggerEndDateTime` still has the previous value instead of `null`.

**Comment by anidotnet** [comment]
This is working as designed, where `null` value is meant to ignore the field during update. To achieve what you want you need to first remove the object and then insert it back with proper values.

**Comment by Try4W** [comment]
Can I insert a document to the specific position in the repository to "replace" the old one?
Or it is better to introduce ordering by creation date?

**Comment by anidotnet** [comment]
See all the collections are a map internally, so there is no concept of position in the main collection.

**Comment by Try4W** [comment]
Okay, I understand.
But I still think that for ObjectsRepository it's expected behavior to replace the existed value with null rather than ignore nulls.

**Comment by anidotnet** [comment]
This limitation is there because in the end and an `ObjectRepository` is just a typed version of `NitriteCollection` and an object in it is just a `Document`.

**Comment by anidotnet** [comment]
On a second thought, I think you are right. There are 2 variants of update api.

```java
WriteResult update(ObjectFilter filter, Document update);
WriteResult update(ObjectFilter filter, Document update, boolean justOnce);
```
and
```java
WriteResult update(T element);
WriteResult update(T element, boolean upsert);
WriteResult update(ObjectFilter filter, T update);
WriteResult update(ObjectFilter filter, T update, boolean upsert);
```
maybe we can use the document variant for updating objects for some specific fields (which is mentioned in the document) and use object variant to update all fields, thus honoring `null` values.

Let me know what you think.

**Comment by Try4W** [comment]
Yep, because for me it is expected that "object" is something solid and not just a "patch" for a database.

It is also sad that when I trying to update an entry via proposed solution, I get two events from the repository listener:
```
Removed
Inserted
```
while in fact there is only one logical update.

One of my friends says that MongoDB allows updating nulls(undefined) in DB as regular value.
Is it a really good choice to ignore them in nitrite?

**Comment by anidotnet** [comment]
No need for special method, `NitriteCollection` already support replace using `update` method.  I'll make object variant of `update` method to replace all fields whatever be the value and document variant of `update` method replace only those fields which are mentioned in the document. I think that will serve the purpose.

**Comment by anidotnet** [comment]
This change is available in 2.2.0-SNAPSHOT now. Let me know if it works as you expected.
